xcm/xcm-executor/src/traits/should_execute.rs

```diff
@@ -16,7 +16,19 @@
 
 use frame_support::traits::ProcessMessageError;
 use sp_std::result::Result;
-use xcm::latest::{Instruction, MultiLocation, Weight};
+use xcm::latest::{Instruction, MultiLocation, Weight, XcmHash};
+
+/// Properyies of an XCM message and its imminent execution.
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct Properties {
+	/// The amount of weight that the system has determined this
+	/// message may utilize in its execution. Typically non-zero only because of prior fee
+	/// payment, but could in principle be due to other factors.
+	pub weight_credit: Weight,
```

Comment:
Since we're putting weight credits into properties, the next natural question to ask is whether we should put max weight into properties as well. It appears to me that the answer is yes, as it does seem to be a property of the message itself, rather than a property of the XCVM.